lib/ansible/modules/cloud/ovirt/ovirt_snapshot.py

```diff
@@ -263,19 +274,33 @@ def main():
 
     vm_service = vms_service.vm_service(vm.id)
     snapshots_service = vms_service.vm_service(vm.id).snapshots_service()
-    try:
-        state = module.params['state']
-        if state == 'present':
-            ret = create_snapshot(module, vm_service, snapshots_service)
-        elif state == 'restore':
-            ret = restore_snapshot(module, vm_service, snapshots_service)
-        elif state == 'absent':
-            ret = remove_snapshot(module, vm_service, snapshots_service)
-        module.exit_json(**ret)
-    except Exception as e:
-        module.fail_json(msg=str(e), exception=traceback.format_exc())
-    finally:
+    if module.params.get('keep_days_old') is not None:
```

Comment:
In this case we wouldn't close the connection. Please create new method something like `remove_old_snapshosts` and call it from state=='present' branch as it's default state. Also please document what will be returned in case this paramter is used.